File: algorithms/[231]2 的幂/solution.ts
// topics = ["位运算"]

/**
 * Bit Manipulation
 * time O(1), space O(1)
 */
function isPowerOfTwo(n: number): boolean {
  if (n <= 0) {
    return false;
  }
  return (n & -n) === n;
}

File: algorithms/[231]2 的幂/solution.py
# topics = ["位运算"]


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        """
        Bit Manipulation
        time O(1), space O(1)
        """
        if n <= 0:
            return False
        return n & (n - 1) == 0
